### PR TITLE
[DataFactory]: Add CMK Properties

### PR DESCRIPTION
**Pull request by Pacodel**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by Pacodel** [comment]
Adding new properties for CMK with UserAssigned Identities scenario, as well as 2 new options in Identity Enum.
This is expected to offer the feature and has minimum impact for existing customers as all are optional.

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json [hunk 1]
```diff
@@ -5046,9 +5084,11 @@
       "properties": {
         "type": {
           "type": "string",
-          "description": "The identity type. Currently the only supported type is 'SystemAssigned'.",
+          "description": "The identity type.",
           "enum": [
-            "SystemAssigned"
+            "SystemAssigned",
+            "UserAssigned",
+            "SystemAssigned,UserAssigned"
```
Comment (shawnxzq) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json:5091]: This is added by accident?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json [hunk 2]
```diff
@@ -5066,12 +5106,28 @@
...
       ]
     },
+    "UserAssignedIdentitiesDefinitionSpecification": {
+      "description": "Definition of all user assigned identities for a factory.",
+      "type": "object",
+      "additionalProperties": {
+        "$ref": "#/definitions/UserAssignedIdentitySpecification"
+      }
+    },
+    "UserAssignedIdentitySpecification": {
```
Comment (shawnxzq) [inline specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json:5126]: Why we have the two types "UserAssignedIdentitiesDefinitionSpecification" & "UserAssignedIdentitySpecification" and no explicit properties defined in them?

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json (hunk 1 above)
Comment (Pacodel) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json:5091]: No, these are valid options for Identity type

File: specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json (hunk 2 above)
Comment (Pacodel) [reply specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/datafactory.json:5126]: It's a dictionary. I copied it from CosmosDB swagger
UserAssignedIdentities is an Object and the properties that it contains it's a Dictionar<string, Object> that's why we need both things. 

"UserAssignedIdentities": {
  "/subscriptions/b3ee3a7f-7614-4644-ad07-afa832620b4b/resourceGroups/ADF/providers/Microsoft.ManagedIdentity/userAssignedIdentities/PacodelTestMI": {},
  "/subscriptions/b3ee3a7f-7614-4644-ad07-afa832620b4b/resourceGroups/ADF/providers/Microsoft.ManagedIdentity/userAssignedIdentities/PacoTestMI2": {}
}

**Review by JeffreyRichter — APPROVED** [review]

**Pull request merged**